src/Cake.Common/IO/Net/HttpAliases.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using Cake.Core;
+using Cake.Core.Annotations;
+using Cake.Core.Diagnostics;
+using Cake.Core.IO;
+
+namespace Cake.Common
+{
+    /// <summary>
+    /// Contains functionality related to HTTP operations
+    /// </summary>
+    [CakeAliasCategory("HTTP Operations")]
+    public static class HttpAliases
+    {
+        /// <summary>
+        /// Downloads the specified File over HTTP to the specified local output path
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="url">URL of file to download.</param>
+        /// <param name="outputPath">Where to download the file to.</param>
+        [CakeMethodAlias]
+        public static void DownloadFile(this ICakeContext context, string url, FilePath outputPath)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+            if (url == null)
+            {
+                throw new ArgumentNullException("url");
+            }
+            if (outputPath == null)
+            {
+                throw new ArgumentNullException("outputPath");
+            }
+
+            context.Log.Write(Verbosity.Verbose, LogLevel.Information, "Downloading File: {0}", url);
+
+            // We track the last posted value since the event seems to fire many times for the same value
+            var percentComplete = 0;
+
+            var http = new System.Net.WebClient();
```

Comment:
As we're using this in a sync fashion we should wrap in a using block, so streams/responses gets deallocated.